### ISSUE TITLE
Migrate analysis pipeline to read from eval metadata instead of model name strings

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/cruijff_kit#391]
## Context

We currently inject structured metadata into `.eval` files at scaffold time:

```bash
--metadata epoch="0"
--metadata finetuned="true"
--metadata source_model="Llama-3.2-3B-Instruct"
```

But nothing in the analysis pipeline reads it yet. Instead, `report_generator.py` and `viz_helpers.py` parse `log.eval.model` (a descriptive string like `hf/Llama-3.2-3B-Instruct_original_epoch_0`) for display names, epoch numbers, and grouping keys.

This works but is fragile — it depends on:
1. The `--model hf/descriptive_name` + `-M model_path` trick (which broke in inspect-ai 0.3.180, see #347)
2. `HF_HUB_OFFLINE=1` to prevent Hub lookups on the fake model name
3. String parsing conventions that aren't enforced anywhere

Since we're already writing the metadata, we should eventually read from it too.

## Proposed change

Update analysis consumers to prefer `log.eval.metadata` fields over `log.eval.model` string parsing:

| Data point | Current source | New source |
|---|---|---|
| Display name | Parse `log.eval.model` | `metadata["source_model"]` |
| Epoch | Parse from model name | `metadata["epoch"]` |
| Fine-tuned? | Infer from path | `metadata["finetuned"]` |
| Task variant | `task_args["vis_label"]` | *(no change — already clean)* |

### Files to update
- `tools/inspect/report_generator.py`
- `tools/inspect/viz_helpers.py` (dedup, display_name, risk plots)
- Skill docs for analyze-experiment

### Backward compat
Fall back to current string-parsing behavior when metadata fields are missing (older `.eval` files).

## Why

- Removes dependence on the `--model` naming trick and `HF_HUB_OFFLINE` workaround
- Makes the pipeline resilient to future inspect-ai changes in model resolution
- We're already paying the cost of writing the metadata — may as well use it

## Priority

Low — current approach works. This is cleanup for long-term robustness.

## Related
- #255 — Original metadata schema design (closed, metadata injection implemented)
- #347 — inspect-ai 0.3.180 broke model name trick, fixed with `HF_HUB_OFFLINE`
- #385 — vis_label dedup fix (task_args side, unaffected by this change)

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

Parking this pending the #539 spike (`run → summarize (strict) → explore`), but not closing it — the disposition genuinely depends on how the spike lands.

The two named target files here (`report_generator.py`, `viz_helpers.py`) are part of the deterministic analyze machinery #539 may retire, and since inspect-viz is unlikely to feature in the spike, `viz_helpers` in particular probably doesn't survive in its current form. So "migrate *these files* to read metadata" may quietly become moot.

But the underlying *principle* — read model/epoch identity from `log.eval.metadata` rather than parsing the `hf/fake_name_epoch_0` model string — outlives the plotting library. Any strict-summarize or `explore` log-read path will still need to know which model produced a log, and should get it from metadata, not string-parsing (which already broke once on the inspect-ai 0.3.180 `--model` trick, #347).

So: holding here as a **design input to #539's log-read layer** rather than a standalone migration. If the spike doesn't land and we keep deterministic analysis, this reopens as-written.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

**Re-scope needed before this is actionable** (flagging while it's fresh — surfaced during the Structural Integrity milestone triage).

The post-run analysis pipeline was restructured in #539 / #548, which makes parts of this issue's body stale:

- **`report_generator.py` was deleted.** Its report-assembly logic is gone; only the PDF-prep helpers survive (now `pdf_preprocess.py`). So it's no longer a file to update here.
- **`analyze-experiment` → `explore-experiment`.** The analysis step is now a Claude-authored exploration, not a deterministic report generator, so the "analyze-experiment skill docs" target needs to point at the explore-experiment docs instead.

**The core idea still holds and is still worth doing:** the remaining string-parsing consumer is **`viz_helpers.py`** (dedup / `display_name` / grouping keys), which still reads the descriptive `log.eval.model` string for epoch / source_model / finetuned instead of the structured `--metadata` we already inject. Migrating it removes the `--model hf/fake_name` trick + `HF_HUB_OFFLINE=1` workaround that broke once already (#347).

Suggested refresh before scheduling: update the file list to `viz_helpers.py` + the explore-experiment docs, drop the `report_generator.py` row, and re-confirm the backward-compat fallback against the current `.eval` metadata schema.

Held out of the Structural Integrity milestone until the body is refreshed.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

Closing as completed during the project archive triage. The migration this asked for landed under other PRs: `report_generator.py` is gone, and `viz_helpers.py` now reads `log.eval.metadata` for `epoch`, `is_finetuned`, and `source_model` (the grouping keys). 

Residual, for the record: the model string is still used as a display label (with a `vis_label` override) and as `model_name` in `parse_eval_log.py`. Metadata is the source of truth; the string is cosmetic.

—MxC
